cpp/src/parquet/column_writer.cc

```diff
@@ -1405,6 +1408,25 @@ class TypedColumnWriterImpl : public ColumnWriterImpl, public TypedColumnWriter<
     return pages_change_on_record_boundaries_;
   }
 
+  void AddKeyValueMetadata(
+      const std::shared_ptr<const KeyValueMetadata>& key_value_metadata) override {
+    if (closed_) {
+      throw ParquetException("Cannot add key-value metadata to closed column");
+    }
```

Comment:
nit: should we throw if `key_value_metadata == nullptr`?

Reply:
in file_writer. the code looks like below:

```c++
  void AddKeyValueMetadata(
      const std::shared_ptr<const KeyValueMetadata>& key_value_metadata) override {
    if (key_value_metadata_ == nullptr) {
      key_value_metadata_ = key_value_metadata;
    } else if (key_value_metadata != nullptr) {
      key_value_metadata_ = key_value_metadata_->Merge(*key_value_metadata);
    }
  }
```

Maybe check null or ignore null is both ok for me

Reply:
I think you're looking at the old code. The code below is the same as file_writer.